SalesforceCommerceCloud/pwa-kit
# @W-14591221 - fix babel config issue where *.ts files can't be imported in ssr.js

# Description

Enhances the `@salesforce/pwa-kit-dev` command to accept option arguments `--babelArgs` which can look like this allowing users of `pwa-kit-dev` to set their own babel CLI options at build time

```diff
-"start": "pwa-kit-dev start",
+"start": "pwa-kit-dev start --babelArgs=\"--extensions '.ts,.tsx,.js,.jsx' --otherBabelArgs '...' \"",
```

Props to @clavery for the babel config tip 🏅 

Fixes #1591 

### Before

```
Error: Cannot find module './typescript-file'
Require stack:
- /path/to/pwa-kit/packages/template-typescript-minimal/app/ssr.js
...
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/path/to/pwa-kit/packages/template-typescript-minimal/app/ssr.js'
  ]
}

Node.js v18.15.0
error: Command failed: /path/to/pwa-kit/packages/pwa-kit-dev/node_modules/.bin/babel-node  /path/to/pwa-kit/packages/template-typescript-minimal/app/ssr.js
```

### After

Works ✨ 





# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- (change1)

# How to Test-Drive This PR

- run `node ~/path/to/local/pwa-kit/packages/pwa-kit-create-app/scripts/create-mobify-app-dev.js --outputDir pwa-kit-minimal-2023-12-05 --preset typescript-minimal-test-project --templateVersion 3.3.0-dev
- navigate to `pwa-kit-minimal-2023-12-05` dir
- run `npm start`
- verify that the app (which now has mixed JS / TS imports originating in `ssr.js`) works






# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
